compiler/testdata/coroutine_durable.go

```diff
@@ -3196,6 +3196,46 @@ func varArgs(_fn0 ...int) {
 		}
 	}
 }
+
+//go:noinline
+func ReturnNamedValue() (_fn0 int) {
+	_c := coroutine.LoadContext[int, any]()
+	var _f0 *struct {
+		IP int
+		X0 int
+	} = coroutine.Push[struct {
+		IP int
+		X0 int
+	}](&_c.Stack)
+	if _f0.IP == 0 {
+		*_f0 = struct {
+			IP int
+			X0 int
+		}{}
+	}
+	defer func() {
+		if !_c.Unwinding() {
+			coroutine.Pop(&_c.Stack)
+		}
+	}()
+	switch {
+	case _f0.IP < 2:
+		_f0.X0 = 5
+		_f0.IP = 2
+		fallthrough
+	case _f0.IP < 3:
+		coroutine.Yield[int, any](11)
+		_f0.IP = 3
+		fallthrough
+	case _f0.IP < 4:
+		_f0.X0 = 42
+		_f0.IP = 4
+		fallthrough
+	case _f0.IP < 5:
+		return _f0.X0
+	}
+	return
```

Comment:
Just saw this. I think this is added by the check that ensures the function ends with a `return`. https://github.com/stealthrocket/coroutine/blob/7c96207066e86683cd621b9de4061b69589dfb00/compiler/compile.go#L635-L646

This probably means either this check needs to be performed before the renaming, or it needs to know about the selectors mapping to generate the right complete return statement.

Reply:
Isn't this statement unreachable tho? `_f0.IP` will never be set to `5`

Reply:
Agreed. Updated the change, replacing the default `return` with a `panic("unreachable")` to validate this stays true.